src/vcpkg/base/system.mac.cpp

```diff
@@ -0,0 +1,262 @@
+#include <vcpkg/base/hash.h>
+#include <vcpkg/base/parse.h>
+#include <vcpkg/base/system.mac.h>
+#include <vcpkg/base/system.process.h>
+
+#include <map>
+
+#if !defined(_WIN32)
+#include <net/if.h>
+#if defined(__APPLE__)
+#include <ifaddrs.h>
+
+#include <net/if_dl.h>
+#define AF_TYPE AF_LINK
+#elif defined(__linux__)
+#include <ifaddrs.h>
+
+#include <netpacket/packet.h>
+#define AF_TYPE AF_PACKET
+#else
+// Fallback to use ioctl calls for systems without getifaddrs()
+#include <unistd.h>
+
+#include <netinet/in.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#endif
+#endif
+
+namespace vcpkg
+{
+    static constexpr size_t MAC_BYTES_LENGTH = 6;
+    // 6 hex-bytes (12 hex-digits) + five separators
+    static constexpr size_t MAC_STRING_LENGTH = MAC_BYTES_LENGTH * 2 + 5;
+
+    bool validate_mac_address_format(StringView mac)
+    {
+        if (mac.size() != MAC_STRING_LENGTH) return false;
+        for (size_t i = 0; i < MAC_STRING_LENGTH; ++i)
+        {
+            // every third character is a ':' separator
+            if (((i + 1) % 3) == 0)
+            {
+                if (mac[i] != ':') return false;
+                continue;
+            }
+
+            if (!ParserBase::is_hex_digit(mac[i])) return false;
+        }
+        return true;
```

Comment:
I think that you need a comment to explain the `%` suggests that this could be clearer. How about one of these?

```
if (mac.size() != MAC_STRING_LENGTH
  || !ParserBase::is_hex_digit(mac[0])
  || !ParserBase::is_hex_digit(mac[1])) { return false; }

for (std::size_t idx = 2; idx < MAC_STRING_LENGTH; idx += 3) { 
  if (':' != mac[idx]
  || !ParserBase::is_hex_digit(mac[idx + 1])
  || !ParserBase::is_hex_digit(mac[idx + 2])) { return false; }
}

return true;
```

or


```
static constexpr char form[] = "XX:XX:XX:XX:XX:XX";
if (mac.size() != MAC_STRING_LENGTH) { return false; }
for (std::size_t idx = 0; idx < MAC_STRING_LENGTH; ++idx) {
    if (form[idx] == ':') {
        if (mac[idx] != ':') { return false; }
    } else {
        if (!ParserBase::is_hex_digit(mac[idx])) { return false; }
    }
}

return true;
```